core/block/editor/state/state.go

```diff
@@ -164,11 +165,11 @@ func (s *State) RootId() string {
 }
 
 func (s *State) NewState() *State {
-	return &State{parent: s, blocks: make(map[string]simple.Block), rootId: s.rootId, noObjectType: s.noObjectType, migrationVersion: s.migrationVersion, uniqueKeyInternal: s.uniqueKeyInternal, originalCreatedTimestamp: s.originalCreatedTimestamp}
+	return &State{parent: s, blocks: make(map[string]simple.Block), rootId: s.rootId, noObjectType: s.noObjectType, migrationVersion: s.migrationVersion, uniqueKeyInternal: s.uniqueKeyInternal, originalCreatedTimestamp: s.originalCreatedTimestamp, notifications: s.CopyNotifications()}
```

Comment:
I think you can optimize things by not copying notifications. If you will use getter for notification so all should be okay. This getter should traverse parents to find the first non-nil notifications map like it did in other parts of state

Reply:
I think the problem here is that the parent state and the state can refer to the same map, which can lead to unexpected behavior or changes, so that's why I use copy here.

Reply:
it's common pattern in state, for example:
```
func (s *State) Get(id string) (b simple.Block) {
	if b = s.blocks[id]; b != nil {
		return
	}
	if s.parent != nil {
		if b = s.Pick(id); b != nil {
			b = b.Copy()
			s.blocks[id] = b
			return
		}
	}
	return
}
```

we defer copying to the moment of write

Reply:
Fixed